common/transport/amqp/devdoc/receiver_link_requirements.md

```diff
@@ -0,0 +1,98 @@
+# ReceiverLink Requirements
+
+## Overview
+
+The `ReceiverLink` class is internal to the SDK and shall be considered private. It shall not be used directly by clients of the SDK, who should instead rely on the higher-level constructs (`Client` classes provided by `azure-iot-device` and `azure-iothub`).
+
+The `ReceiverLink` class implements a state machine that manages the underlying `amqp10` link object used to receive messages from IoT Hub. It can be attached and detached manually, and will try to attach automatically if not already attached when setting up a message listener or trying to settle an existing message.
+
+## Example usage
+
+```typescript
+import * as amqp10 from 'amqp10';
+import { AmqpMessage } from './amqp_message';
+
+const linkAddress = 'exampleAddress';
+const amqp10Client = new amqp10.AmqpClient(null);
+
+const receiverLink = new ReceiverLink(linkAddress, null, amqp10Client);
+receiverLink.on('error', (err) => {
+  console.error(err.toString());
+});
+
+receiverLink.on('message', (msg) => {
+  console.log(msg.body);
+  receiverLink.accept(msg, (err) => {
+    if (err) {
+      console.error(err.toString());
+    } else {
+      console.log('message successfully settled (accept)');
+    }
+  });
+});
+```
+
+## Public Interface
+
+### constructor(linkAddress: string, linkOptions: any, amqp10Client: amqp10.AmqpClient)
+
+**SRS_NODE_AMQP_RECEIVER_LINK_16_002: [** The `ReceiverLink` class shall inherit from `EventEmitter`. **]**
+
+**SRS_NODE_AMQP_RECEIVER_LINK_16_003: [** The `ReceiverLink` class shall implement the `AmqpLink` interface. **]**
+
+### attach(callback: (err?: Error) => void): void
```

Comment:
>callback [](start = 11, length = 8)

is callback required?

Reply:
yeah I'd rather make callbacks mandatory on all our lower-level APIs and only make them optional on the outer layer, public APIs. Since this is a new API, this is by design.

---
In reply to: [139490418](https://github.com/Azure/azure-iot-sdk-node/pull/83#discussion_r139490418) [](ancestors = 139490418)